R/DBConnection.R

```diff
@@ -106,6 +106,39 @@ setGeneric("dbSendQuery",
   valueClass = "DBIResult"
 )
 
+#' @export
+dbBreak <- structure(list(), class = "dbi_break_sentinel")
+
+dbFetchChunked <- function(rs, n, callback) {
+  while (TRUE) {
+    chunk <- dbFetch(rs, n = n)
+    if (nrow(chunk) == 0) {
+      return(chunk)
+    }
+    result <- callback(df = chunk)
+    if (identical(result, dbBreak)) {
```

Comment:
How about catching conditions of a specific class, instead of relying on a sentinel?

```
  continue_loop <- tryCatch({ result <- callback(...); TRUE }, dbi_abort = function(e) FALSE)
```